SimExLite/PMICalculators/SimpleScatteringPMICalculator.py

```diff
@@ -0,0 +1,11147 @@
+#        {
+#            printf( "% e % e % e\n" , rotmat[ii][jj*3+0] ,  rotmat[ii][jj*3+1] ,
+#                                      rotmat[ii][jj*3+2] ) ;
+#        }
+#        printf("\n") ;
+
+#        for ( jj=0 ; jj < 3 ; jj++ )
+#        {
+#            printf( "% e % e % e\n" , invmat[ii][jj*3+0] ,  invmat[ii][jj*3+1] ,
+#                                      invmat[ii][jj*3+2] ) ;
+#        }
+#        printf("\n\n\n") ;
+#    }
+# }
+
+
+##############################################################################
+
+
+def s2e_rand_orient(r, mat):
+
+    N = r.shape[0]
+    ###    print N
+    vv = numpy.zeros((3, 0))
+    for ii in range(N):
+        vv = r[ii, :]
+        ###        print vv , vv.shape , mat.shape
+        r[ii, 0] = mat[0] * vv[0] + mat[1] * vv[1] + mat[2] * vv[2]
+        r[ii, 1] = mat[3] * vv[0] + mat[4] * vv[1] + mat[5] * vv[2]
+        r[ii, 2] = mat[6] * vv[0] + mat[7] * vv[1] + mat[8] * vv[2]
+
+
+##############################################################################
+def f_eval_numE(a_snp, a_sample):
+
+    num_Z = len(list(a_sample["selZ"].keys()))
+    all_numE = numpy.zeros((num_Z,))
+    cc = 0
+    for sel_Z in list(a_sample["selZ"].keys()):
+        all_numE[cc] = numpy.mean(a_snp["q"][a_sample["selZ"][sel_Z]])
+        cc = cc + 1
+    return all_numE
+
+
+##############################################################################
+
+
+def f_md_step(r, v, m, dt):
+    r = r.copy()
+    v = v.copy()
+    a = (force(config, param) + external_force(config, param)) / m
+    r = r + v * dt + 0.5 * a * dt**2.0
+    config["r"] = r
+    an = (force(config, param) + external_force(config, param)) / m
+    v = v + 0.5 * (a + an) * dt
+    config["v"] = v
+    E = f_sysenergy_kin(v, m) + syspot(config, param) + ext_syspot(config, param)
+
+
+##############################################################################
+
+
+def f_pmi_diagnostics_help():
+    print(
+        """
+    ----
+    """
+    )
+
+    ##############################################################################
+    ##############################################################################
+    ##############################################################################
+
+
+def load_ff_database():
+
+    # INTERNAL NOTE #  ########    To get to the nice array below:    ########
+    # INTERNAL NOTE #   for x in `seq 1 100` ; do echo $x ; RES=`time xatom -Z $x  -formfactor -Q 10 -N_Q 100 2>ff-$x.err` ; echo "$RES" | grep -v \# > ff-$x.dat ; echo "$RES" | grep \# > ff-$x.log ; done
+    # INTERNAL NOTE #   cp ff-1.dat ff_all.dat
+    # INTERNAL NOTE #   for x in `seq 2 100` ; do cat ff-${x}.dat| cut -c14-26 | paste -d" " ff_all.dat - > ff_all.xxx ; mv ff_all.xxx ff_all.dat ; done
+    # INTERNAL NOTE #   cat ff_all.dat |while read x; do echo '[' `echo $x|sed s/\ /\ ,\ /g` '] ,'; done > ff_all.dat--numpy_array
+
+    dbase = numpy.array(
+        [
+            [
+                0.00000000,
```

Comment:
This should not be exploded like this... Black can be toggled for a specific part of code via
```
[code that is formatted]
# fmt: off
[code that won't be formatted]
# fmt: on
[code formatted again]
```

Reply:
It's a very good point. The code format is comments are added. I also agree with the idea of "data and code should be separated". However, as the codes were inherited from the previous eucall project, and probably will not be updated in the foreseeable future, I would just leave it as it is at this moment.